homeassistant/components/broadlink/switch.py

```diff
@@ -3,22 +3,106 @@
 import logging
 
 from broadlink.exceptions import BroadlinkException
+import voluptuous as vol
 
-from homeassistant.components.switch import SwitchDeviceClass, SwitchEntity
+from homeassistant.components.switch import (
+    PLATFORM_SCHEMA,
+    SwitchDeviceClass,
+    SwitchEntity,
+)
 from homeassistant.const import (
     CONF_COMMAND_OFF,
     CONF_COMMAND_ON,
+    CONF_FRIENDLY_NAME,
+    CONF_HOST,
+    CONF_MAC,
     CONF_NAME,
+    CONF_SWITCHES,
+    CONF_TIMEOUT,
+    CONF_TYPE,
     STATE_ON,
     Platform,
 )
+import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.restore_state import RestoreEntity
 
 from .const import DOMAIN
 from .entity import BroadlinkEntity
+from .helpers import data_packet, import_device, mac_address
 
 _LOGGER = logging.getLogger(__name__)
 
+CONF_SLOTS = "slots"
+
+SWITCH_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME): cv.string,
+        vol.Optional(CONF_COMMAND_OFF): data_packet,
+        vol.Optional(CONF_COMMAND_ON): data_packet,
+    }
+)
+
+OLD_SWITCH_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_COMMAND_OFF): data_packet,
+        vol.Optional(CONF_COMMAND_ON): data_packet,
+        vol.Optional(CONF_FRIENDLY_NAME): cv.string,
+    }
+)
+
+PLATFORM_SCHEMA = vol.All(
+    cv.deprecated(CONF_HOST),
+    cv.deprecated(CONF_SLOTS),
+    cv.deprecated(CONF_TIMEOUT),
+    cv.deprecated(CONF_TYPE),
+    PLATFORM_SCHEMA.extend(
+        {
+            vol.Required(CONF_MAC): mac_address,
+            vol.Optional(CONF_HOST): cv.string,
+            vol.Optional(CONF_SWITCHES, default=[]): vol.Any(
+                cv.schema_with_slug_keys(OLD_SWITCH_SCHEMA),
+                vol.All(cv.ensure_list, [SWITCH_SCHEMA]),
+            ),
+        }
+    ),
+)
```

Comment:
We can remove the deprecation warnings and the old platform schema. This could be rewritten as:
```python3
PLATFORM_SCHEMA = vol.Schema(
    {
        vol.Required(CONF_MAC): mac_address,
        vol.Optional(CONF_HOST): cv.string,
        vol.Optional(CONF_SWITCHES, default=[]): vol.All(
            cv.ensure_list, [SWITCH_SCHEMA],
        ),
    }
)
```